internal/io/limitwriter.go

```diff
@@ -0,0 +1,49 @@
+// Copyright The Notary Project Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package io provides a LimitWriter that writes to an underlying writer up to
+// a limit.
+
+package io
+
+import "io"
+
+// LimitWriter is a writer that writes to an underlying writer up to a limit.
+type LimitWriter struct {
+	w       io.Writer
+	limit   int64
+	written int64
+}
+
+// NewLimitWriter returns a new LimitWriter that writes to w.
+//
+// parameters:
+// w: the writer to write to
+// limit: the maximum number of bytes to write
+func NewLimitWriter(w io.Writer, limit int64) *LimitWriter {
+	return &LimitWriter{w: w, limit: limit}
+}
+
+// Write writes p to the underlying writer up to the limit.
+func (lw *LimitWriter) Write(p []byte) (int, error) {
+	remaining := lw.limit - lw.written
+	if remaining <= 0 {
+		return 0, nil
```

Comment:
This should return an error. `ErrShortWrite` might be a proper error.

Reply:
Updated.